### PR TITLE
HHH-15664 Fix array section in documentation

### PR DESCRIPTION
**Pull request by beikov**
https://hibernate.atlassian.net/browse/HHH-15664

### CODE REVIEW

File: documentation/src/main/asciidoc/userguide/chapters/domain/collections.adoc [hunk 1]
```diff
@@ -1049,51 +1049,59 @@ include::{extrasdir}/collections-map-bidirectional-example.sql[]
...
-Hibernate does support the mapping of arrays in the Java domain model - conceptually the same as mapping a List.
-However, it is important to realize that it is impossible for Hibernate to offer lazy-loading for arrays of entities and, for this reason,
-it is strongly recommended to map a "collection" of entities using a List rather than an array.
+NOTE: It is impossible for Hibernate to offer lazy-loading for arrays of entities and, for this reason,
+it is strongly recommended to map a "collection" of entities using a List or Set rather than an array.
 
-[[collections-array-binary]]
-==== Arrays as binary
+[[collections-array-as-basic]]
+==== Arrays as basic value type
```
Comment (yrodiere) [inline documentation/src/main/asciidoc/userguide/chapters/domain/collections.adoc:1058]: Please don't break anchors :( People linking to `/current/` or `/stable/` would get what essentially amounts to a dead link.

If you really want to change the anchor, something like this should work (assuming you're using the latest version of Asciidoctor):

```suggestion
[[collections-array-as-basic]]
==== [[collections-array-binary]] Arrays as basic value type
```

File: documentation/src/main/asciidoc/userguide/chapters/domain/collections.adoc [hunk 2]
```diff
@@ -1049,51 +1049,59 @@ include::{extrasdir}/collections-map-bidirectional-example.sql[]
...
+By default, Hibernate will choose a type for the array, based on `Dialect.getPreferredSqlTypeCodeForArray()`.
+Prior to Hibernate 6.1, the default was to always use the BINARY type, as supported by the current `Dialect`,
+but now, Hibernate will leverage the native array data types if possible.
 
-[[collections-array-binary-example]]
-.Arrays stored as binary
+To force the BINARY type, the persistent attribute has to be annotated with `@JdbcTypeCode(SqlTypes.VARBINARY)`.
+
+[[collections-array-as-basic-example]]
+.Arrays stored as SQL array
```
Comment (yrodiere) [inline documentation/src/main/asciidoc/userguide/chapters/domain/collections.adoc:1067]:
```suggestion
[[collections-array-binary-example]] To force the BINARY type, the persistent attribute has to be annotated with `@JdbcTypeCode(SqlTypes.VARBINARY)`.

[[collections-array-as-basic-example]]
.Arrays stored as SQL array
```

File: documentation/src/main/asciidoc/userguide/chapters/domain/collections.adoc [hunk 3]
```diff
@@ -1049,51 +1049,59 @@ include::{extrasdir}/collections-map-bidirectional-example.sql[]
...
+NOTE: It is impossible for Hibernate to offer lazy-loading for arrays of entities and, for this reason,
+it is strongly recommended to map a "collection" of entities using a List or Set rather than an array.
 
-[[collections-array-binary]]
-==== Arrays as binary
+[[collections-array-as-basic]]
+==== Arrays as basic value type
 
-By default, Hibernate will choose a BINARY type, as supported by the current `Dialect`.
+By default, Hibernate will choose a type for the array, based on `Dialect.getPreferredSqlTypeCodeForArray()`.
```
Comment (yrodiere) [inline documentation/src/main/asciidoc/userguide/chapters/domain/collections.adoc:1060]: Found the native german speaker :)

```suggestion
By default, Hibernate will choose a type for the array based on `Dialect.getPreferredSqlTypeCodeForArray()`.
```

**Review by yrodiere — COMMENTED** [review]
Thanks. Here are my two cents...

**Review by sebersole — COMMENTED** [review]
Aside from Yoann's comments, lgtm

**Pull request merged**